pkg/asset/config/appliance_config.go

```diff
@@ -138,63 +149,137 @@ func (a *ApplianceConfig) Load(f asset.FileFetcher) (bool, error) {
-	// Note: in Load func, we ensure that CpuArchitecture is not nil and fallback to x86_64
-	return swag.StringValue(a.Config.OcpRelease.CpuArchitecture)
+func (a *ApplianceConfig) validateApiVersion() field.ErrorList {
+	if a.Config.TypeMeta.APIVersion == "" {
+		return field.ErrorList{field.Required(field.NewPath("apiVersion"), "appliance-config version is required")}
+	}
+	switch v := a.Config.APIVersion; v {
+	case types.ApplianceConfigVersion: // Current version
+	default:
+		return field.ErrorList{field.Invalid(field.NewPath("apiVersion"),
+			a.Config.TypeMeta.APIVersion,
+			fmt.Sprintf("appliance-config version must be %q", types.ApplianceConfigVersion))}
+	}
+	return nil
 }
 
-func GetReleaseArchitectureByCPU(arch string) string {
-	switch arch {
-	case CpuArchitectureX86:
-		return ReleaseArchitectureAMD64
-	case CpuArchitectureAARCH64:
-		return ReleaseArchitectureARM64
-	default:
-		return arch
+func (a *ApplianceConfig) validateOcpRelease() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate ocpRelease.version
+	if a.Config.OcpRelease.Version == "" {
+		allErrs = append(allErrs, field.ErrorList{field.Required(field.NewPath("ocpRelease.version"),
+			"ocpRelease version is required")}...)
+	}
+	minOcpVer, _ := version.NewVersion(MinOcpVersion)
+	ocpVer, err := version.NewVersion(a.Config.OcpRelease.Version)
+	if err != nil {
+		allErrs = append(allErrs, field.ErrorList{field.Invalid(field.NewPath("ocpRelease.version"),
+			a.Config.OcpRelease.Version,
+			fmt.Sprintf("OCP release version must be in major.minor or major.minor.patch format %q", err))}...)
+
+	}
+	if ocpVer.LessThan(minOcpVer) {
+		allErrs = append(allErrs, field.ErrorList{field.Invalid(field.NewPath("ocpRelease.version"),
+			a.Config.OcpRelease.Version,
+			fmt.Sprintf("OCP release version must be at least %s", MinOcpVersion))}...)
+	}
+
+	// Validate ocpRelease.channel
+	if swag.StringValue(a.Config.OcpRelease.Channel) != "" {
+		switch graph.ReleaseChannel(*a.Config.OcpRelease.Channel) {
+		case graph.ReleaseChannelStable:
+		case graph.ReleaseChannelFast:
+		case graph.ReleaseChannelCandidate:
+		case graph.ReleaseChannelEUS:
+		default:
+			allErrs = append(allErrs, field.ErrorList{field.Invalid(field.NewPath("ocpRelease.channel"),
+				a.Config.OcpRelease.Channel,
+				"specified OCP release channel is not supported ")}...)
+		}
 	}
+
+	// Validate ocpRelease.cpuArchitecture
+	if swag.StringValue(a.Config.OcpRelease.CpuArchitecture) != "" {
+		switch *a.Config.OcpRelease.CpuArchitecture {
+		case CpuArchitectureX86:
+		case CpuArchitectureAARCH64:
+		case CpuArchitecturePPC64le:
+		default:
+			allErrs = append(allErrs, field.ErrorList{field.Invalid(field.NewPath("ocpRelease.cpuArchitecture"),
+				a.Config.OcpRelease.CpuArchitecture,
+				"specified OCP release cpuArchitecture is not supported ")}...)
+		}
+	}
+
+	return allErrs
+}
+
+func (a *ApplianceConfig) validateDiskSize() error {
+	if a.Config.DiskSizeGB < MinDiskSize {
```

Comment:
note: we need to see what happens with float values

Reply:
Fails on json unmarshaling, maybe we can make it nicer later:
`FATAL failed to load asset "Appliance Config": failed to unmarshal appliance-config.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal number 200.2 into Go struct field ApplianceConfig.diskSizeGb of type int`